core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -467,7 +474,7 @@ private[spark] class TaskSchedulerImpl private[scheduler](
       taskState: TaskState,
       reason: TaskFailedReason): Unit = synchronized {
     taskSetManager.handleFailedTask(tid, taskState, reason)
-    if (!taskSetManager.isZombie && taskState != TaskState.KILLED) {
+    if (!taskSetManager.isZombie) {
```

Comment:
This shouldn't be a behavior change, just a simplification of the logic to always call reviveOffers(). Whether the task is rescheduled or not is decided independently of this call -- it's just nice to call reviveOffers always so the task won't end up pending until the next round of offers.